src/engine/engine.js

```diff
@@ -35,21 +35,44 @@ function createState(initState) {
   };
```

Comment:
This was wrong here. We shouldn't keep the state separately from the store. The update method has quite a substantial performance impact (I checked it). We just have to keep the state here.